File: backend/src/main/java/com/example/demo/service/MessageService.java
package com.example.demo.service;

import com.example.demo.model.Message;
import com.example.demo.model.Order;
import com.example.demo.model.User;

import java.util.List;

public interface MessageService {
    Message sendMessage(User sender, User recipient, Order order, String content);
  //  List<Message> getMessagesForOrder(Order order);
    List<Message> getMessagesForOrder(Order order, String recipientUsername, String senderUsername);
    List<Message> getDialogMessages(Order order, String user1, String user2);
    List<Message> getAdminUserDialogMessages(Order order, String user, String admin);
}

File: frontend/src/Services/MessageService.js
import ApiService from './BaseService/ApiService';

class MessageService {
  sendMessage(message) {
    return ApiService.post('/messages', message);
  }

  getMessagesForOrder(orderId, recipientUsername, senderUsername) {
    return ApiService.get(`/messages/${orderId}?recipientUsername=${recipientUsername}&senderUsername=${senderUsername}`);
  }
  getAdminUserDialogMessages(orderId, user) {
    return ApiService.get(`/messages/${orderId}/admin-dialog?user=${user}`);
  }
  getDialogMessages(orderId, user1, user2) {
    return ApiService.get(`/messages/${orderId}/dialog?user1=${user1}&user2=${user2}`);
  }
}

export default new MessageService();